src/walberla_bridge/src/lattice_boltzmann/LBWalberlaImpl.hpp

```diff
@@ -398,6 +426,24 @@ class LBWalberlaImpl : public LBWalberlaBase {
 #endif
   }
 
+  void setup_streaming_communicator() {
+    auto const setup = [this]<typename PdfPackInfo>() {
+      auto const &blocks = m_lattice->get_blocks();
+      m_pdf_streaming_communicator =
+          std::make_shared<PDFStreamingCommunicator>(blocks);
+      m_pdf_streaming_communicator->addPackInfo(
+          std::make_shared<PdfPackInfo>(m_pdf_field_id));
+      m_pdf_streaming_communicator->addPackInfo(
+          std::make_shared<PackInfoStreaming<VectorField>>(
+              m_last_applied_force_field_id));
+    };
+    if (m_has_boundaries or (m_collision_model and has_lees_edwards_bc())) {
+      setup.template operator()<PackInfo<PdfField>>();
+    } else {
+      setup.template operator()<PackInfoStreaming<PdfField>>();
+    }
```

Comment:
The code-generated `PackInfo` do not work with UBB. After two days of investigation, I still don't understand why there is extra fluid densities flowing into LB fluid cells that are in contact with LB flag cells that have a slip velocity. You can replace `PackInfo` by `PackInfoStreaming` at line 441 and run `./pypresso ../testsuite/python/lb_boundary_velocity.py LBBoundaryVelocityTest.test_wall_slip_parallel` to reproduce the bug. Edit the collide and stream kernels to print the values streamed along directions 5 and 6 (top resp. bottom) in the first fluid cell in contact with a flag cell. Use a box size of `[8, 1, 1]` in units of agrid for simplicity.